scheds/rust/scx_lavd/src/bpf/sys_stat.bpf.c

```diff
@@ -213,17 +220,41 @@ static u64 clamp_time_slice_ns(u64 slice)
 static void calc_sys_stat(struct sys_stat_ctx *c)
 {
 	static int cnt = 0;
-	u64 avg_svc_time = 0, cur_sc_util;
+	u64 avg_svc_time = 0, cur_sc_util, scu_spike;
 
+	/*
+	 * Calculate the CPU utilization.
+	 */
 	c->duration_total = c->duration * nr_cpus_onln;
 	c->compute_total = time_delta(c->duration_total, c->idle_total);
 	c->cur_util = (c->compute_total << LAVD_SHIFT) / c->duration_total;
 
 	cur_sc_util = (c->tot_sc_time << LAVD_SHIFT) / c->duration_total;
 	if (cur_sc_util > c->cur_util)
 		cur_sc_util = min(sys_stat.avg_sc_util, c->cur_util);
-	c->cur_sc_util = cur_sc_util;
 
+	/*
+	 * When calculating the scaled utilization, which determines the
+	 * required compute capacity and the number of active CPUs, we add the
+	 * half-capacities of CPUs where utilization spikes occurred.
+	 *
+	 * Hence, when a CPU is almost fully utilized (say 90%) during a
+	 * period, we may underestimate the utilization. For example, when the
+	 * measured CPU utilization is 100%, there is a possibility that the
+	 * actual utilization is actually higher, such as 130%.
```

Comment:
It's not very clear to me how the utilization can be > 100%, maybe you can clarify this a bit more in this comment?

Reply:
Oh I see, that's because of capacity comes into play, it's clear now, thanks!